crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -662,30 +712,30 @@ protected void wait(BulkExecutionContext ctx) {
         }
     }
 
-    protected Callable<Response> getFutureRequest(final Request req) {
+    protected Callable<Response> getFutureRequest(final Request req, RequestMetrics metrics) {
         return new Callable<Response>() {
             @Override
             public Response call() {
                 LOGGER.debug("Starting a future {} request",req.getOperation());
                 switch (req.getOperation()) {
                     case FIND:
-                        return find((FindRequest) req);
+                        return find((FindRequest) req, metrics);
                     case INSERT:
-                        return insert((InsertionRequest) req);
+                        return insert((InsertionRequest) req, metrics);
                     case DELETE:
-                        return delete((DeleteRequest) req);
+                        return delete((DeleteRequest) req, metrics);
                     case UPDATE:
-                        return update((UpdateRequest) req);
+                        return update((UpdateRequest) req, metrics);
                     case SAVE:
-                        return save((SaveRequest) req);
+                        return save((SaveRequest) req, metrics);
                     default:
                         throw new UnsupportedOperationException("CRUD operation '"+req.getOperation()+"' is not supported!");
                 }
             }
         };
     }
 
-    public BulkResponse bulkRequest(BulkRequest requests) {
+    public BulkResponse bulkRequest(BulkRequest requests, RequestMetrics metrics) {
         LOGGER.debug("Bulk request start");
         Error.push("bulk operation");
         ExecutorService executor = Executors.newFixedThreadPool(factory.getBulkParallelExecutions());
```

Comment:
Might be fine to track the whole bulk request still, without any other information other than it being a bulk request. There isn't really anything else we can add in a sane way I think.